From 5375414fa4ccfa49db28374c31ca48eb153b655d Mon Sep 17 00:00:00 2001
From: Saliba-san <65911528+Saliba-san@users.noreply.github.com>
Date: Wed, 1 May 2024 16:54:32 -0300
Subject: [PATCH] hwdb: Add mapping for Samsung GalaxyBook - 550X (#32616)

- Kernel: linux-6.8.8-arch
- systemd: 255.5-3-arch
- Laptop: Samsung GalaxyBook - NP550XDA-KF1BR

This is the solution to a problem with the Samsung GalaxyBook in which some Media keys,
when pressed and released, get stuck as if they are hold.

(cherry picked from commit a6ab3053aab515ecae7568e0beefee7dbe6f9100)

Related: RHEL-5950
---
 hwdb.d/60-keyboard.hwdb | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/hwdb.d/60-keyboard.hwdb b/hwdb.d/60-keyboard.hwdb
index 2e1838744b..20430c9085 100644
--- a/hwdb.d/60-keyboard.hwdb
+++ b/hwdb.d/60-keyboard.hwdb
@@ -1750,6 +1750,11 @@ evdev:atkbd:dmi:bvn*:bvr*:bd*:svn[sS][aA][mM][sS][uU][nN][gG]*:pn*550P*:*
  KEYBOARD_KEY_a8=!                                      # Fn Lock - Function lock on
  KEYBOARD_KEY_a9=!                                      # Fn Lock - Function lock off
 
+evdev:atkbd:dmi:bvn*:bvr*:bd*:svn[sS][aA][mM][sS][uU][nN][gG]*:pn*550X*:*
+ KEYBOARD_KEY_ce=!prog1                                 # Fn+F1 launch settings
+ KEYBOARD_KEY_ae=!volumedown                            # Fn+F7 volume up
+ KEYBOARD_KEY_b0=!volumeup                              # Fn+F8 volume down
+
 # Series 7 / 9
 evdev:atkbd:dmi:bvn*:bvr*:bd*:svn[sS][aA][mM][sS][uU][nN][gG]*:pn*350V*:*
 evdev:atkbd:dmi:bvn*:bvr*:bd*:svn[sS][aA][mM][sS][uU][nN][gG]*:pn*670Z*:*
